src/bio/bio_buffer.c

```diff
@@ -605,7 +642,8 @@ dma_map_one(struct bio_desc *biod, struct bio_iov *biov, void *arg)
 	last_rg = iod_last_region(biod);
 
 	/* First, try consecutive reserve from the last reserved region */
-	if (last_rg) {
+	if (last_rg && bio_iov2media(biov) != DAOS_MEDIA_SCM &&
+	    bio_iov2media(biov) == last_rg->brr_media) {
```

Comment:
just curious that is it possible bio_iov2media(biov) != last_rg->brr_media?

Reply:
Yes, that's quite possible. For example, a fetch request to fetch [0, 4k), on VOS storage, the [0, 4k) could consist of [0, 1k)@epoch 2 on SCM and [1k, 4k)@epoch 1 on NVMe.